Develop

mramazanov opened the pull request:
2. Создана конфигурация для rabbitmq
3. Создан слушатель для очереди rabbitmq
4. Создан метод для отрисовки символьной 2D графики

mramazanov committed:
1. Созданы модели
2. Создана конфигурайия для rabbitmq
3. Создан слушатель для очереди rabbitmq
4. Создан метод для отрисовки символьной 2D графики

gmnpr commented on src/main/java/com/javajabka/x6_notification/listener/NotificationConsumer.java:
```diff
@@ -0,0 +1,24 @@
+import org.springframework.stereotype.Component;
+
+@Log4j2
+@Component
```
почему в сервисе order у тебя логика отправки на уровне сервиса, а тут на уровне компонента? мне кажется будет полезно все это унифицировать

gmnpr commented on src/main/java/com/javajabka/x6_notification/Utils/AsciiArtUtil.java:
```diff
@@ -0,0 +1,38 @@
+import java.awt.image.BufferedImage;
+
+@Component
+public class AsciiArtUtil {
```
Util класс должен отвечать определнными требованиям, наприме он не должен быть бином, у него не должно быть конструктора, а методы должны быть статичными

gmnpr replied:
отлично! мы почти там где надо
далее нужно сделать приватных конструктор (мы же не хотим, чтобы кто-то создал экземпляр класса утилиты)

ИЛИ

кинуть анотацию UtilityClass (от лумбук), тогда, вроде, даже методы не обязательно делать статическими (лумбук все сделает сам)

gmnpr commented on src/main/java/com/javajabka/x6_notification/Utils/AsciiArtUtil.java:
```diff
@@ -0,0 +1,38 @@
+            if (sb.toString().trim().isEmpty()) {
+                continue;
+            }
+            System.out.println(sb);
```
так логируют только маленькие жава жабки, мы уже знаем log.info!

gmnpr commented on src/main/resources/application.properties:
```diff
@@ -1 +1,11 @@
-spring.application.name=x6-notification
\ No newline at end of file
+spring.application.name=x6-notification
```
а почему 2 строки запись занимает, там фантомные пробелы?)

mramazanov committed:
1. Убрана аннотация компонента для утилитарного класса AsciiArtUtil и исправлен метод на статический
2. Изменен возвращаемый тип для метода класса AsciiArtUtil
3. Продьюсер для rabbimq в сервисе Order вынесен в компонент

gmnpr commented on src/main/java/com/javajabka/x6_notification/listener/NotificationConsumer.java:
```diff
@@ -0,0 +1,17 @@
+    @RabbitListener(queues = "${app.rabbitmq.queue}")
+    public void receive(OrderResponse orderResponse) {
+        log.info(AsciiArtUtil.getAsciiArtMessage());
+        log.info("Received message: " + orderResponse);
```
самый правильный вариант написания выглядит так 
log.info("Received message: {}", orderResponse);

mramazanov replied:
Сделал изначально так, но подумал что усложняю))

gmnpr commented on src/main/java/com/javajabka/x6_notification/Utils/AsciiArtUtil.java:
```diff
@@ -0,0 +1,40 @@
+package com.javajabka.x6_notification.Utils;
```
видимо пакет очень уважаемый раз он с большой буквы)))

gmnpr commented on src/main/java/com/javajabka/x6_notification/model/OrderResponse.java:
```diff
@@ -0,0 +1,13 @@
+import java.util.List;
+
+@Data
+@Builder
```
а вот билдеры в этих моделях они реально нужны?)

mramazanov committed:
1. Убрал аннотацию билдера для моделей
2. Изменил на правильный метод логировния логера
3. Исправил имя пакета utils

gmnpr commented on src/main/java/com/javajabka/x6_notification/listener/NotificationConsumer.java:
```diff
@@ -0,0 +1,17 @@
+public class NotificationConsumer {
+    @RabbitListener(queues = "${app.rabbitmq.queue}")
+    public void receive(OrderResponse orderResponse) {
+        log.info("{}", AsciiArtUtil.getAsciiArtMessage());
```
тут все было норм...

mramazanov committed:
1. Добавление lombok аннотации для утилитного класса AsciiArtUtil
2. Использование метода логировния ascii-art сообщения без саплаера
